File: duplicate-zeros/duplicate-zeros.cpp
class Solution {
public:
    void duplicateZeros(vector<int>& nums) {
        int n = nums.size();
        for(int i=0;i<n;i++) {
            if(nums[i]==0) {
                for(int j = nums.size()-1;j>i;j--) {
                    nums[j] = nums[j-1];
                }
               nums[i++] = 0;
            }
        }
    }
};

File: duplicate-zeros/duplicate-zeros.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function(arr) {
    const n = arr.length;
    for (let i = 0;i<n ;i++) {
        if(arr[i] ==0) {
            for(let j = n-1; j > i; j--) {
                arr[j] = arr[j-1];
            }
            arr[i++] = 0;
        }
    }
};